packages/wallet-sdk/src/CoinbaseWalletProvider.ts

```diff
@@ -42,17 +42,27 @@ export class CoinbaseWalletProvider extends EventEmitter implements ProviderInte
   private readonly handlers = {
     fetch: (request: RequestArguments) => fetchRPCRequest(request, this.chain),
 
-    sign: (request: RequestArguments) => {
-      if (request.method === 'eth_requestAccounts') {
-        if (this.connected) return this.accounts;
-      } else {
-        if (!this.connected) {
-          throw standardErrors.provider.unauthorized(
-            "Must call 'eth_requestAccounts' before other methods"
-          );
-        }
+    // eth_requestAccounts
+    handshake: async (_: RequestArguments): Promise<AddressString[]> => {
+      if (this.connected) return this.accounts;
+      const accounts = await this.signHandler.handshake();
+      this.emit('connect', { chainId: prepend0x(this.chain.id.toString(16)) });
+      return accounts;
```

Comment:
moved from SignRequestHandler
![image](https://github.com/coinbase/coinbase-wallet-sdk/assets/966662/f1b769bc-8a94-4eea-b0fc-00f944a50222)

Reply:
we need to put handshake() into a try block as well

on FE side, user can click sign out from anywhere, including eth_requestAccounts step, when that happens, FE will respond an unauthorized error back to SDK, and we need to catch it here and call disconnect to clean up